xnat_tools/bids_utils.py

```diff
@@ -134,6 +134,44 @@ def prepare_heudiconv_output_path(
     return heudi_output_dir
 
 
+def bidsmap_scans(scans, bidsmap = None):
+    """Filter the series descriptions based on the bidsmap file"""
+    # NOTE (BNR): We could break these down into smaller functions, one for
```

Comment:
I think it's also an okay balance. If for some reason we need to start doing extra checks that get more complex we can revisit